webapp/components/suggestion/at_mention_provider.jsx

```diff
@@ -126,22 +127,35 @@ function filterUsersByPrefix(users, prefix, limit, type) {
 }
 
 export default class AtMentionProvider {
-    handlePretextChanged(suggestionId, pretext) {
+    handlePretextChanged(suggestionId, pretext, argChannelId) {
```

Comment:
@hmhealey 

> It makes it much easier to use this in places that aren't just the main post textbox, but instead of passing the channelId as an argument to handlePretextChanged, could you try passing it into the constructor for the AtMentionProvider when it's created instead?

I agree that is much better and tried it first but unfortunately it didn't work; The channel id in `AtMentionProvider` was not updated when the user changes the post to reply.

In my observation, `AtMentionProvider` belongs to a textbox and is created only if the textbox is created, but the textbox seems to be resued when the user changes the post to reply.